torch/autograd/gradcheck.py

```diff
@@ -1526,16 +1526,18 @@ def gradgradcheck(
     tupled_inputs = _as_tuple(inputs)
 
     if grad_outputs is None:
-        # If grad_outputs is not specified, create random Tensors of the same
-        # shape, type, and device as the outputs
-        def randn_like(x):
-            y = torch.randn_like(
-                x if (x.is_floating_point() or x.is_complex()) else x.double(), memory_format=torch.legacy_contiguous_format)
-            if gen_non_contig_grad_outputs:
-                y = torch.testing.make_non_contiguous(y)
-            return y.requires_grad_()
+        # If grad_outputs is not specified, create random Tensors of the same shape, type, and device as the outputs
+        def make_tensor_like(x):
+            return torch.testing.make_tensor(
```

Comment:
How is this one different from the previous? 
In particular, things like layout or memory format are not copied anymore?

Reply:
> How is this one different from the previous?

By default `make_tensor` gives you uniform values `(-10, 10)` so we should have better coverage there. 

Note that this change is not to improve the gradcheck, but rather to deprecate the undocumented and untested `make_non_contiguous` from `torch.testing`.

> In particular, things like layout or memory format are not copied anymore?

- `layout` was not really functional in the first place. For example: 

    ```py
    >>> t = torch.randn(1, 2, 3).to_sparse_coo()
    >>> torch.randn_like(t)
    NotImplementedError: Could not run 'aten::normal_' with arguments from the 'SparseCPU' backend. [...]
    ```
- `memory_format` is not copied from the input tensor, but rather fixed to `torch.legacy_contiguous_format`. `make_tensor` generates contiguous tensors by default so we don't need to pass anything here.

Reply:
> By default make_tensor gives you uniform values (-10, 10) so we should have better coverage there.

Not sure what you mean by better coverage?
There are some subtleties in the numerical precision for these functions and I would prefer the values to remain in the [-1, 1] range if possible.

> Note that this change is not to improve the gradcheck, but rather to deprecate the undocumented and untested make_non_contiguous from torch.testing.

Sure! That's not problem! I just want to make sure we're not changing the behavior of gradcheck in the process.

Reply:
> Not sure what you mean by better coverage?

Better coverage of different values. Happy to change `[-1, 1]` if that is required here. 

Just wondering why used a normal distribution before if we want to have this interval. Without this patch we already have about 30% of the values outside the interval.

Reply:
I don't think there is a very strong argument for using normal vs uniform really.
The thing is that we end up doing reduction on pretty big matrices and so the increased variance can be detrimental for precision.
We do have some tests that are quite close to the numerical threshold so I don't want to start introducing flakyness.

Reply:
Thanks for the update!